### PR TITLE
RFC: usb: device: Allow limited amount of tries for write

### LINKED ISSUE #16223: stm32: Unable to send 64 byte packet over control endpoint
**Describe the bug**
cdc_acm_composite does not work on olimexino_stm32. Reverting commit e5cbe6a9e7 (usb: cdc: Add unique PIDs for each sample) fixes the problem.

 
**To Reproduce**

1. compile ```subsys/usb/cdc_acm_composite``` sample
```
cd samples/subsys/usb/cdc_acm_composite
rm -rf build && mkdir build && cd build
cmake -DBOARD=olimexino_stm32  ..
make
```
2. burn zephyr image into olimexino_stm32;
3. connect olimexino_stm32 to linux host via USB;
4. reset olimexino_stm32;
5. No ```/dev/ttyACMx``` device registered on linux host.
See ```dmesg``` output on linux host (some details may vary):
```
usb 2-1.1: new full-speed USB device number 102 using ehci-pci
usb 2-1.1: New USB device found, idVendor=2fe3, idProduct=0002, bcdDevice= 0.11
usb 2-1.1: New USB device strings: Mfr=1, Product=2, SerialNumber=3
usb 2-1.1: can't set config #1, error -32
```

**Expected behavior**
Registering two new ```/dev/ttyACMx``` devices is expected on linux host.
Expected ```dmesg``` output on linux host (some details may vary):
```
usb 2-1.1: new full-speed USB device number 101 using ehci-pci
usb 2-1.1: New USB device found, idVendor=2fe3, idProduct=0100, bcdDevice= 0.11
usb 2-1.1: New USB device strings: Mfr=1, Product=2, SerialNumber=3
usb 2-1.1: Product: Zephyr CDC ACM sample
usb 2-1.1: Manufacturer: ZEPHYR
usb 2-1.1: SerialNumber: 0.01
cdc_acm 2-1.1:1.0: ttyACM0: USB ACM device
cdc_acm 2-1.1:1.2: ttyACM1: USB ACM device
```

**Environment (please complete the following information):**

USB Host OS: Debian Linux x86_64, kernel version 4.18.6-1
Toolchain: Zephyr SDK 0.10.0
Zephyr git describe: zephyr-v1.14.0-836-gfc3270d09c

### PR DESCRIPTION
**Pull request by finikorg**
Set limited amount of tries to write to USB control endpoint EP0.

### CODE REVIEW

**Review by carlescufi — COMMENTED** [review]
Can you please add a more precise explanation in the commit message?

**Comment by masz-nordic** [comment]
Number of retries could be a `Kconfig` value.

**Comment by finikorg** [comment]
> Number of retries could be a `Kconfig` value.

This is kind of workaround, I think it should not be in Kconfig

**Comment by jfischer-phytec-iot** [comment]
This would probably fix one of the bugs I found in request handling, I will open a issue.

File: subsys/usb/usb_device.c [hunk 1]
```diff
@@ -1160,15 +1155,14 @@ int usb_disable(void)
...
+	int ret;
 
-		if (ret != -EAGAIN) {
-			return ret;
-		}
+	do {
+		ret = usb_dc_ep_write(ep, data, data_len, bytes_ret);
+	} while (ret == -EAGAIN && tries--);
 
-		k_yield();
```
Comment (jfischer-phytec-iot) [inline subsys/usb/usb_device.c]: I prefer to keep k_yield or k_sleep here.

File: subsys/usb/usb_device.c [hunk 2]
```diff
@@ -1160,15 +1155,14 @@ int usb_disable(void)
 
 int usb_write(u8_t ep, const u8_t *data, u32_t data_len, u32_t *bytes_ret)
 {
-	while (true) {
-		int ret = usb_dc_ep_write(ep, data, data_len, bytes_ret);
+	int tries = 10;
```
Comment (jfischer-phytec-iot) [inline subsys/usb/usb_device.c]: Number of tries should have Kconfig option, maybe range 3 10?

**Review by jfischer-no — CHANGES_REQUESTED** [review]

**Comment by ioannisg** [comment]
Moving to next release since the review was not finalized for v.2.0.0

File: include/kernel.h [hunk 3]
```diff
@@ -2918,7 +2920,42 @@ extern void k_delayed_work_init(struct k_delayed_work *work,
...
+ *
+ * @warning
+ * A delayed work item must not be modified until it has been processed
+ * by the workqueue.
+ *
+ * @note Can be called by ISRs.
+ *
+ * @param work_q Address of workqueue.
+ * @param work Address of delayed work item.
+ * @param delay Delay before submitting the work item (in microseconds).
```
Comment (dbkinder) [inline include/kernel.h]: actual param name is ``us_delay`` in the code below

File: boards/xtensa/up_squared_adsp/doc/index.rst [hunk 4]
```diff
@@ -0,0 +1,4 @@
+.. _Up_Squared_Audio_DSP:
+
+Up Squared Audio DSP
+####################
```
Comment (dbkinder) [inline boards/xtensa/up_squared_adsp/doc/index.rst]: need some doc content here , but you know that :)

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Pull request closed without merging**